nbqa/replace_source.py

```diff
@@ -32,28 +34,37 @@ def main(python_file: "Path", notebook: "Path") -> None:
 
     pycells = pyfile[len(CODE_SEPARATOR) :].split(CODE_SEPARATOR)
 
-    def _reinstate_magics(source: str) -> List[str]:
+    def _reinstate_magics(
+        source: str, trailing_semicolons: List[int], cell_number: int
+    ) -> List[str]:
         """
         Put (commented-out) magics back in.
 
         Parameters
         ----------
         source
             Portion of Python file between cell separators.
+        trailing_semicolons
+            List of cells which originally had trailing semicolons.
+        cell_number
+            Number of current cell.
 
         Returns
         -------
         List[str]
             New source that can be saved into Jupyter Notebook.
         """
+        if cell_number in trailing_semicolons:
+            source = source.rstrip("\n") + ";"
```

Comment:
actually, this risks adding _two_ semicolons if the command doesn't remove semicolons...I should first check there isn't already a trailing comma

Reply:
wrote failing test, fixed now